pkg/controller/conditions.go

```diff
@@ -28,7 +28,41 @@ import (
 type VirtualMachineConditionManager struct {
 }
 
-func (d *VirtualMachineConditionManager) CheckFailure(vmi *v1.VirtualMachineInstance, syncErr error, reason string) (changed bool) {
+func NewVirtualMachineConditionManager() *VirtualMachineConditionManager {
+	return &VirtualMachineConditionManager{}
+}
+
+func (d *VirtualMachineConditionManager) GetCondition(vm *v1.VirtualMachine, cond v1.VirtualMachineConditionType) *v1.VirtualMachineCondition {
+	if vm == nil {
+		return nil
+	}
+	for _, c := range vm.Status.Conditions {
+		if c.Type == cond {
+			return &c
+		}
+	}
+	return nil
+}
+
+func (d *VirtualMachineConditionManager) HasCondition(vm *v1.VirtualMachine, cond v1.VirtualMachineConditionType) bool {
+	return d.GetCondition(vm, cond) != nil
+}
+
+func (d *VirtualMachineConditionManager) RemoveCondition(vm *v1.VirtualMachine, cond v1.VirtualMachineConditionType) {
+	var conds []v1.VirtualMachineCondition
+	for _, c := range vm.Status.Conditions {
+		if c.Type == cond {
+			continue
+		}
+		conds = append(conds, c)
+	}
+	vm.Status.Conditions = conds
+}
+
+type VirtualMachineInstanceConditionManager struct {
```

Comment:
Is there a need to have this as struct? Could be this only simple functions?

Reply:
I used the same design as the other structs in this file use, to be consistent. I agree that the structs are useless, we could remove them and rename all the functions in a new PR. Function overloading would be ideal for this use case, but Go does not support it.